frontend/viewer/src/lib/components/field-editors/rich-multi-ws-input.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+  import {type IRichMultiString, type IWritingSystem} from '$lib/dotnet-types';
+  import type {ReadonlyDeep} from 'type-fest';
+  import {Label} from '../ui/label';
+  import LcmRichTextEditor from '../lcm-rich-text-editor/lcm-rich-text-editor.svelte';
+  import type {IRichString} from '$lib/dotnet-types/generated-types/MiniLcm/Models/IRichString';
+
+  let {
+    value = $bindable(),
+    ...constProps
+  }: {
+    value: IRichMultiString;
+    readonly?: boolean;
+    writingSystems: ReadonlyArray<ReadonlyDeep<IWritingSystem>>;
+    onchange?: (wsId: string, value: IRichString, values: IRichMultiString) => void;
+    autofocus?: boolean;
+  } = $props();
+
+  const {
+    readonly = false,
+    writingSystems,
+    onchange,
+    autofocus,
+  } = $derived(constProps);
+
+  function setValue(newValue: IRichString, wsId: string) {
+    newValue?.spans.forEach((span) => span.ws ??= wsId);
+    value[wsId] = newValue;
+    onchange?.(wsId, value[wsId], value);
```

Comment:
With moving onchange to be called here means it fires on keypress and not onblur now. We should probably just be doing this work on in change anyway.
Also we should probably write tests for this issue.